PR for BXMSDOC-8435: Update the release notes docs for RHPAM/RHDM 7.12.1 (#133)
kaldesai/kie-docs: doc-content/enterprise-only/release-notes/rn-7.12.1-fixed-issues-ref.adoc

== Build and assembly

* In Spring Framework, versions 5.3.0 - 5.3.10, 5.2.0 - 5.2.17, and an older unsupported versions, it is possible for a user to provide malicious input to cause the insertion of additional log entries [https://issues.redhat.com/browse/RHPAM-4098[RHPAM-4098]]

ifdef::PAM[]

== {PROCESS_ENGINE_CAP}

* Boundary timer is not triggered after a rollback [https://issues.redhat.com/browse/RHPAM-4102[RHPAM-4102]]
* Incorrect response for REST service when `org.kie.server.bypass.auth.user` system property is used with `JAASUserGroupCallbackImpl` implementation [https://issues.redhat.com/browse/RHPAM-4087[RHPAM-4087]]
* When you use `ClusteredJobFailOverListener` method, you receive system errors caused by `ConcurrentModificationException` exception [https://issues.redhat.com/browse/RHPAM-4069[RHPAM-4069]]
* Certificate based authentication is required between {KIE_SERVER} and process instance migration tool [https://issues.redhat.com/browse/RHPAM-4038[RHPAM-4038]]
* Smart router is sending requests to a stopped {KIE_SERVER} [https://issues.redhat.com/browse/RHPAM-4035[RHPAM-4035]]

> * The `ProcessServiceImpl.getAvailableSignals` must obtain process instance in read-only mode to avoid race condition [https://issues.redhat.com/browse/RHPAM-3967[RHPAM-3967]]

Reviewer: ```suggestion
* `ProcessServiceImpl.getAvailableSignals` must obtain process instances in read-only mode to avoid race condition [https://issues.redhat.com/browse/RHPAM-3967[RHPAM-3967]]
```

> * The `ProcessServiceImpl.getAvailableSignals` must obtain process instance in read-only mode to avoid race condition [https://issues.redhat.com/browse/RHPAM-3967[RHPAM-3967]]

Reviewer: Ideally would be an adjective so something like The ProcessServiceImpl.getAvailableSignals` method. I don't know what  ProcessServiceImpl.getAvailableSignals` so I have removed the article.

Author: Okay noted! It is a method.

* The `OptimisticLockRetryInterceptor` triggers a retry logic when `javax.persistence.OptimisticLockException` or `org.hibernate.exception.ConstraintViolationException` exceptions are thrown. This impacts the pluggable variable persistence when persisting the custom entity classes [https://issues.redhat.com/browse/RHPAM-3948[RHPAM-3948]]
* All the scheduled notifications are duplicated on restart [https://issues.redhat.com/browse/RHPAM-3946[RHPAM-3946]]
* Even after the timers are fired, entries from `EJBTimerScheduler.localCache` are not removed [https://issues.redhat.com/browse/RHPAM-3936[RHPAM-3936]]
* In a setup with multiple {KIE_SERVER} nodes, duplicate EJB timer instances are created [https://issues.redhat.com/browse/RHPAM-3929[RHPAM-3929]]

> * Unable to find the task instance with the ID `xxx`, if the audit mode is disabled [https://issues.redhat.com/browse/RHPAM-3847[RHPAM-3847]]

Reviewer: ```suggestion
* Unable to find the task instance with the ID `xxx` if  audit mode is disabled [https://issues.redhat.com/browse/RHPAM-3847[RHPAM-3847]]
```

== Process Designer

* When you make a rest call to retrieve the task form where a data object contains a `LocalDate` type, it is not displayed on any browsers. An input box displays an empty value [https://issues.redhat.com/browse/RHPAM-4020[RHPAM-4020]]

> * If there are multiple `intermediateCatchEvent` events, you receive an `ExtensibleXmlParser` error due to duplicate ID value [https://issues.redhat.com/browse/RHPAM-3878[RHPAM-3878]]

Reviewer: ```suggestion
* If there are multiple `intermediateCatchEvent` events, you receive an `ExtensibleXmlParser` error due to duplicate ID values [https://issues.redhat.com/browse/RHPAM-3878[RHPAM-3878]]
```

endif::[]

== DMN Designer

* When the same FEEL object-access invocation is used by two KIE containers, you receive DMN evaluation errors [https://issues.redhat.com/browse/RHDM-1877[RHDM-1877]]

== {OPENSHIFT}

> * When you install different version of {OPENSHIFT} operator in each namespace on the same {OPENSHIFT} cluster, you receive validation error and CRD conflict [https://issues.redhat.com/browse/RHPAM-4158[RHPAM-4158]]

Reviewer: ```suggestion
* When you install different versions of the {OPENSHIFT} operator in each namespace on the same {OPENSHIFT} cluster, you receive a validation error and CRD conflict [https://issues.redhat.com/browse/RHPAM-4158[RHPAM-4158]]
```

> * Authorization fails while using the role mapping [https://issues.redhat.com/browse/RHPAM-4146[RHPAM-4146]]

Reviewer: ```suggestion
* Authorization fails while using role mapping [https://issues.redhat.com/browse/RHPAM-4146[RHPAM-4146]]
```

* When you set the default role variable as `AUTH_LDAP_DEFAULT_ROLE`, {OPENSHIFT} images ignore the LDAP roles [https://issues.redhat.com/browse/RHPAM-4132[RHPAM-4132]]

> * DockerImage kind is not accepted in business automation operator with JDBC extension images [https://issues.redhat.com/browse/RHPAM-3787[RHPAM-3787]]

Reviewer: ```suggestion
* The DockerImage type is not accepted in the business automation operator with JDBC extension images [https://issues.redhat.com/browse/RHPAM-3787[RHPAM-3787]]
```

> * The counter `kie_server_container_started_total` displays the total number of containers with status `STARTED` and `DEACTIVATED`, but it counts the `DEACTIVATED` containers as well and gives wrong count [https://issues.redhat.com/browse/RHPAM-3784[RHPAM-3784]]

Reviewer: ```suggestion
* The  `kie_server_container_started_total` counter displays the total number of containers with the status `STARTED` and `DEACTIVATED`, but it counts the `DEACTIVATED` containers as well and gives the wrong count [https://issues.redhat.com/browse/RHPAM-3784[RHPAM-3784]]
```

* Increase the `KieExecutorMDB` threads in {PRODUCT} image [https://issues.redhat.com/browse/RHPAM-3528[RHPAM-3528]]

== Decision engine

* When you use `RuleTerminalNodeLeftTuple`, the delete or insert child facts causes a loop in `PhreakJoinNode.doRightInserts()` function and you receive an `OutOfMemoryError` error [https://issues.redhat.com/browse/RHDM-1865[RHDM-1865]]
* When you are executing the rules multiple times, you receive a random exception in a specific rule after constraint jitting [https://issues.redhat.com/browse/RHDM-1861[RHDM-1861]]

> * You must define `allVars` field as `final` in the `org.mvel2.util.VariableSpaceModel` class [https://issues.redhat.com/browse/RHDM-1854[RHDM-1854]]

Reviewer: ```suggestion
* You must define the `allVars` field as `final` in the `org.mvel2.util.VariableSpaceModel` class [https://issues.redhat.com/browse/RHDM-1854[RHDM-1854]]
```

> * Result type of `BigDecimal` operation with one or more null value operands must not be a `String` or `Boolean` type value [https://issues.redhat.com/browse/RHDM-1852[RHDM-1852]]

Reviewer: ```suggestion
* The result type of the `BigDecimal` operation with one or more null value operands must not be a `String` or `Boolean` type value [https://issues.redhat.com/browse/RHDM-1852[RHDM-1852]]
```
